### ISSUE TITLE
[no-unnecessary-type-assertion] False negative on unknown!

### ISSUE THREAD

**Issue by ooyyloo** [typescript-eslint/typescript-eslint#1082]
**Repro**

[Playground](https://typescript-eslint.io/play#ts=5.3.2&fileType=.tsx&code=KYDwDg9gTgLgBAYwDYEMDOa4EE4G8BQccAJijCgPwBccArgHYDW9EA7vQNz4C%2B%2B%2BokWHHQBPegjgAzBghgBLCPTgxgaGAAoAlDQAKUCAFs5aYAB4GzNvQB8eQnCTB4KGjgC8cesFbYtXIlBOtFBKKAB0pOQAhFy8QA&eslintrc=N4KABGBEBOCuA2BTAzpAXGUEKQAIBcBPABxQGNoBLY-AWhXkoDt8B6Jge1tiacTJTIAhtEK0ipWkOTJE0fJQ5N0YANrhsOOdA7RIGiAF0NAXxAmgA&tsconfig=N4KABGBEDGD2C2AHAlgGwKYCcDyiAuysAdgM6QBcYoEEkJemy0eAcgK6qoDCAFutAGsylBm3TgwAXxCSgA&tokens=false)

```JSON
{
  "rules": {
    "@typescript-eslint/no-unnecessary-type-assertion": ["error"]
  }
}
```

```TS
export class A {
  data?: unknown;
}

export async function test(): Promise<unknown> {
  let a: A = new A();
  return a.data!;
}

```

**Expected Result**
no errors

**Actual Result**
This assertion is unnecessary since it does not change the type of the expression. eslint(@typescript-eslint/no-unnecessary-type-assertion)

**Additional Info**

**Versions**

| package                            | version |
| ---------------------------------- | ------- |
| `@typescript-eslint/eslint-plugin` | `2.3.3` |
| `@typescript-eslint/parser`        | `2.3.3` |
| `TypeScript`                       | `3.6.3` |
| `ESLint`                           | `6.5.1` |
| `node`                             | `12.10.0` |
| `npm`                              | `6.10.3` |

**Comment by bradzacher** [comment]
We don't have explicit support for `unknown`, we should probably add it.
